src/compiler/crystal/syntax/lexer.cr

```diff
@@ -3134,14 +3134,24 @@ module Crystal
       Slice.new(@reader.string.to_unsafe + start_pos, end_pos - start_pos)
     end
 
-    def ident_start?(char)
+    def self.ident_start?(char)
       char.ascii_letter? || char == '_' || char.ord > 0x9F
```

Comment:
Ideally yes. I think the lexer has already ruled out upper case letters as constants at any point where that might be relevant, though. So there should be no practical difference.

Reply:
Or... maybe not: 😆 
```cr
"".@FOO # Error: can't infer the type of instance variable '@FOO' of String
```

Reply:
So we might want to tighten that down. Probably better in a follow-up. `char.ord > 0x9F` looks also *very* permissive.

Reply:
Uppercase instance variable names are actually allowed in Ruby. Here are the other direct uses of `ident_start?` in the lexer:

* Reading a symbol literal. `:FOO` is definitely a valid symbol.
* Reading a percent string literal. This is probably acceptable, because `%Q()` is a valid string literal.
* Reading a global variable. The only ones that can be declared are lib external variables, and the parser rejects ones starting with an uppercase letter.

Uppercase def names are also allowed there. Even in Crystal they are allowed for lib funs, so a `Call` node with an uppercase method name is not necessarily invalid. I don't think we need to do anything here.